### ISSUE TITLE
HTTP server in a language besides Python

### ISSUE THREAD

**Issue by styromaniac** [HelloZeroNet/ZeroNet#2460]
I've played with plenty of HTTP server applications on Android and Linux. They feel much faster than ZeroNet. Now that I've published a copy of the Bible with an htm file per chapter (which means many htm files), ZeroNet is slow as hell with this site.

http://127.0.0.1:43110/1Mjd9zn88kNX1QEkFaq2MrWeMthnR15Fga/

Is there by any chance an Apache server implementation in rust or another close to metal programming language that could be used?

**Comment by purplesyringa** [comment]
Are you using Termux or ZeroNet app from Google Play?

**Comment by HelloZeroNet** [comment]
- Is the page load is slow?
 - Is it also slow if you are using http://127.0.0.1:43110/raw/1Mjd9zn88kNX1QEkFaq2MrWeMthnR15Fga/ ?
 - Is it also slow if you access it using a proxy?  eg. https://zeronet.now.im/raw/1Mjd9zn88kNX1QEkFaq2MrWeMthnR15Fga/

I would recommend to add:
```
 <base href="" target="_top" id="base">
 <script>base.href = document.location.href.replace("/media", "").replace("index.html", "").replace(/[&?]wrapper=False/, "").replace(/[&?]wrapper_nonce=[A-Za-z0-9]+/, "")</script>
```

To right before `</head>`, because without it every link click will result unnecessary redirects, which can be slow.

It could be also possible that android browsers are intentionally slowing down iframes. Not sure if there is any way to avoid that.

**Comment by updatethelicensefuckhead** [comment]
ZeroNet app is a big shit and yes he using Termux you retarded fuck..

**Comment by styromaniac** [comment]
> Are you using Termux or ZeroNet app from Google Play?

Neither

I'm on Linux and using the new Android app. It's universally slow.

**Comment by styromaniac** [comment]
> * Is the page load is slow?
> * Is it also slow if you are using http://127.0.0.1:43110/raw/1Mjd9zn88kNX1QEkFaq2MrWeMthnR15Fga/ ?
> * Is it also slow if you access it using a proxy?  eg. https://zeronet.now.im/raw/1Mjd9zn88kNX1QEkFaq2MrWeMthnR15Fga/
> 
> I would recommend to add:
> 
> ```
>  <base href="" target="_top" id="base">
>  <script>base.href = document.location.href.replace("/media", "").replace("index.html", "").replace(/[&?]wrapper=False/, "").replace(/[&?]wrapper_nonce=[A-Za-z0-9]+/, "")</script>
> ```
> 
> To right before `</head>`, because without it every link click will result unnecessary redirects, which can be slow.
> 
> It could be also possible that android browsers are intentionally slowing down iframes. Not sure if there is any way to avoid that.

It's not just Android. It's also Linux having issues with speed on that site.

**Comment by styromaniac** [comment]
I just now saw other Bibles published on ZeroNet and they're fast. IDK why this site is slow in ZeroNet. It's fast on IPFS. Maybe it's the SCSS or CSS.

**Comment by HelloZeroNet** [comment]
I don't have any problem with the site btw. 
What do you call slow? 
Takes seconds to display the site? 
Does the top-right 0 button appears long before the page load? 
Is it also slow when you reload the same page?

**Comment by styromaniac** [comment]
> I don't have any problem with the site btw.
> What do you call slow?
> Takes seconds to display the site?
> Does the top-right 0 button appears long before the page load?
> Is it also slow when you reload the same page?

It takes several seconds for some chapter pages to load/display even while they were already on the device.

**Comment by styromaniac** [comment]
https://github.com/PR0M3TH3AN/eternal-word/issues/5

Here I opened an issue for The Eternal Word just in case it's not ZeroNet's fault.

**Comment by styromaniac** [comment]
I'm closing this issue because Nofish provided handy code for the fix. It works great for a site I made myself, so I've noticed the improved speed there. The Eternal Word's developer needs to apply that code Nofish provided.

The problem for me is that there are at least 1,189 of these htm files for each of 21 languages (24,969 files), so I think the developer of The Eternal Word needs to use whatever tool they had for editing the heads of all these htm files at the same time in one stroke.

I certainly don't have the lifespan to edit all these htm files one by one and lack the tools to make this actually feasible for myself.

It's appreciated if someone would soft fork The Eternal Word just to add that code to all ~25K files.
